subsys/net/ip/tcp.c

```diff
@@ -2815,11 +2815,17 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	int close_status = 0;
 	enum net_verdict verdict = NET_DROP;
 
-	if (th) {
-		/* Currently we ignore ECN and CWR flags */
-		fl = th_flags(th) & ~(ECN | CWR);
+	if (conn == NULL || pkt == NULL) {
+		NET_ERR("Invalid parameters");
+		return NET_DROP;
 	}
 
+	th = th_get(pkt);
```

Comment:
We should still check if `th` is not NULL after this I think? Theoretically `th_get()` could fail and return NULL.